src/wallet/wallet.cpp

```diff
@@ -4171,3 +4171,54 @@ void CWallet::SetActiveScriptPubKeyMan(uint256 id, OutputType type, bool interna
     spk_man->SetType(type, internal);
     spk_mans[type] = spk_man;
 }
+
+bool CWallet::HasWalletDescriptor(const WalletDescriptor& desc) const
+{
+    if (!IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+        return false;
+    }
+
+    for (auto spk_man_pair : m_spk_managers) {
+        if (spk_man_pair.second->HasWalletDescriptor(desc)) {
```

Comment:
Instead of adding a `HasWalletDescriptor` to `ScriptPubKeyMan`, I think it would be better to use a `dynamic_pointer_cast`. The resulting pointer will be `nullptr` if the object can't be cast to the correct type.

Reply:
So I gave this some thought...
a/ I'm generally not a big fan of dynamic casting. It's an expensive operation (we're in a loop here which might make it worse) and casting doesn't look clean. 

b/ Otoh, you're probably right that the base `ScriptPubKeyMan` class probably shouldn't know about `WalletDescriptor` at all? (although initially I thought legacy wallets can contain descriptors thru `importmulti`)

This being the case, I have no strong preference here. I can switch to dynamic casting.